### PR TITLE
Implemented Symmetry454 chronology

### PR DESCRIPTION
**Pull request by catull**
This is an implementation of Symmetry454 as specified under http://individual.utoronto.ca/kalendis/symmetry.htm.

### CODE REVIEW

**Comment by Clockwork-Muse** [comment]
Quick question: Did you check if the `AccountingChronology` matches the behavior here (and for Symmetry010)?  I think I checked once, and the leap years varied (the cycle was slightly different), but it might be good to look into it again.

**Comment by catull** [comment]
I have to mind-read here, not sure where your question is leading.

Are you asking if the leap-years line up, if the leap year is based on the same formula perhaps ?

The Accounting calendar is based on a cycle of 400 years.
In this cycle, the leap years occur 5, 6 or 7 years apart, giving this delta table:

6 6 5 6 5 
6 6 5 6 5 
6 6 5 6 5 
6 6 
6 6 5 6 5 
6 6 5 6 5 
6 6 5 6 5 
6 6 5 7 5 6 5 
6 6 5 6 5 
6 6 5 6 5 
6 6 5 
6 6 6 5 
6 6 5 6 5 
6 6 5 6 5 
6 6 5 6 5 

The '7' in the middle is no mistake.

The Symmetry454 and Symmetry010 (hence unified as SymmetryXYX) have a cycle of 293 years.
The delta table is:

6 6 5 6 5 
6 6 5 
6 6 5 
6 6 5 6 5 
6 6 5 
6 6 5 
6 6 5 6 5 
6 6 5 
6 6 5 
6 6 5 6 5 
6 6 5 
6 6 5 
6 6 5 6 5 
6 6 5 

If we line up the leap-years of both the Accounting and SymmetryXYX calendars, they are mostly off by 3 and occasionally 2 years.

Starting from the year 1 of Common Era, these are the first leap years:

SymmetryXYX:  3,   9, 15, 20, 26, 31, 37, 43, ...
Accounting     :  6, 12, 17, 23, 28, 34, 40, 45, ...

Not once do the leap years meet.

Now that this is established, are you saying there is no room for either Symmetry454 or Symmetry010 ?
There is room for them, because they serve different purposes.

The Accounting Calendar is "customisable", SymmetryXYX is rigid.
There are, however, structural similarities.

Both calendars group "normal" years with 364 days, and leap years with 371 days.
In other words, each year starts on the same week-day and spans either 52 or 53 full weeks on purpose.

The Accounting Calendar is available with 12 or 13 months.

In the 13-month year each month has exactly 28 days, which has no correspondent in SymmetryXYX.
The 12-month variant features 8 months of 28 days and 4 months with 35 days, which looks a lot like Symmetry454.

An Accounting Calendar may start/end on any month you like, basically.
SymmetryXYX calendars start with January 1st only.

However, the biggest difference is their the leap year formula and occurrence.

**Comment by Clockwork-Muse** [comment]
Hunh, I thought there _were_ some shared leap years.  Guess not (maybe it was an artifact of one of my earlier implementations).
I wasn't saying there was no place for it, especially because I was still pretty sure there were some differences in the leap-year cycle.  But it's good to have this pointed out for others who might be wondering the same.
This is nicely researched!

**Comment by catull** [comment]
I spoke too soon.
It turns out the two cycles intersect.

A couple of common leap years are: 
9284 9684 10084 10456 10484 10856 10884 10980 11256 11273 11284 11380 11397 11628 11656 ...

Serves me right to limit my search to the range of 1 .. 5000.
You were right all along.

**Comment by Clockwork-Muse** [comment]
Heh, not likely to be a factor anytime soon, and that's definitely farther out than I would have checked.

**Comment by catull** [comment]
Any update, do you need pieces of information from me ? Has the code review started yet ?

**Comment by Clockwork-Muse** [comment]
@catull - sorry, I've gotten wrapped up in corefx.  I think you'll have to wait for @jodastephen ...
